### PR TITLE
EVG-13071: GitHub statuses for downstream patches

### PR DESCRIPTION
**Pull request by malikchaya2**
image
https://github.com/evergreen-ci/commit-queue-sandbox/pull/83

### CODE REVIEW

File: trigger/patch.go [hunk 1]
```diff
@@ -203,9 +203,16 @@ func (t *patchTriggers) makeData(sub *event.Subscription) (*commonTemplateData,
...
 		apiModel:          &api,
 		githubState:       message.GithubStatePending,
-		githubContext:     "evergreen",
 		githubDescription: "tasks are running",
 	}
+
+	if t.patch.IsChild() {
+		GitHashLastFour := t.patch.Githash[len(t.patch.Githash)-4:]
+		data.githubContext = fmt.Sprintf("evergreen/%s/%s", t.patch.GithubPatchData.BaseBranch, GitHashLastFour)
+	} else {
```
Comment (ybrill) [inline trigger/patch.go:212]: Another way to do this would be to leave L206 in place and omit this else.

File: trigger/patch.go [hunk 2]
```diff
@@ -203,9 +203,16 @@ func (t *patchTriggers) makeData(sub *event.Subscription) (*commonTemplateData,
 		PastTenseStatus:   t.data.Status,
 		apiModel:          &api,
 		githubState:       message.GithubStatePending,
-		githubContext:     "evergreen",
 		githubDescription: "tasks are running",
 	}
+
+	if t.patch.IsChild() {
+		GitHashLastFour := t.patch.Githash[len(t.patch.Githash)-4:]
+		data.githubContext = fmt.Sprintf("evergreen/%s/%s", t.patch.GithubPatchData.BaseBranch, GitHashLastFour)
```
Comment (ybrill) [inline trigger/patch.go:211]: I don't think the child patch has `GithubPatchData.BaseBranch` populated, though it wouldn't be hard to populate.

Comment (ybrill) [inline trigger/patch.go:211]: did you decide not to go with the child project's name?

Comment (malikchaya2) [reply trigger/patch.go:211]: I wasn't sure if this exists in the patches collection in the db...

File: trigger/patch.go (hunk 1 above)
Comment (malikchaya2) [reply trigger/patch.go:212]: This felt cleaner to me than setting it and then overwriting it. I hear both.

File: trigger/patch.go (hunk 2 above)
Comment (malikchaya2) [reply trigger/patch.go:211]: good point, I'll look into setting it to the project.

Comment (ybrill) [reply trigger/patch.go:211]: We do [set it on the patch](https://github.com/evergreen-ci/evergreen/blob/2f99d8a2ee363999f3d4751e559f0748f408003b/units/patch_intent.go#L710), though the ID might not be what we want these days.

File: trigger/patch.go [hunk 3]
```diff
@@ -206,9 +206,30 @@ func (t *patchTriggers) makeData(sub *event.Subscription) (*commonTemplateData,
...
+		"t.patch.IsChild()": t.patch.IsChild(),
+	})
+
 	if t.patch.IsChild() {
 		GitHashLastFour := t.patch.Githash[len(t.patch.Githash)-4:]
-		data.githubContext = fmt.Sprintf("evergreen/%s/%s", t.patch.GithubPatchData.BaseBranch, GitHashLastFour)
+		data.githubContext = fmt.Sprintf("evergreen/%s/%s", t.patch.Project, GitHashLastFour)
+
+		lastFourPatchID := t.patch.Id.Hex()[len(t.patch.Id.Hex())-4:]
+		grip.Info(message.Fields{
```
Comment (malikchaya2) [inline trigger/patch.go:222]: Please ignore logging, I'll remove it soon

File: trigger/patch.go [hunk 4]
```diff
@@ -203,9 +203,16 @@ func (t *patchTriggers) makeData(sub *event.Subscription) (*commonTemplateData,
 		PastTenseStatus:   t.data.Status,
 		apiModel:          &api,
 		githubState:       message.GithubStatePending,
-		githubContext:     "evergreen",
 		githubDescription: "tasks are running",
 	}
+
+	if t.patch.IsChild() {
+		lastFourPatchID := t.patch.Id.Hex()[len(t.patch.Id.Hex())-4:]
+		data.githubContext = fmt.Sprintf("evergreen/%s/%s", t.patch.Project, lastFourPatchID)
```
Comment (ybrill) [inline trigger/patch.go:211]: I think for new projects the Project field will not be intelligible. Instead, we'd want the project's Identifier.
We could potentially fetch the projectRef with its ID here.

**Review by ybrill — APPROVED** [review]

**Comment by malikchaya2** [comment]
evergreen merge

**Pull request merged**